internal/lint/lint.go

```diff
@@ -171,58 +171,42 @@ func NewLinter(id string, purpose string, addCheck func(func(*text.Failure), str
 
 // GetLinters returns the Linters for the LintConfig.
 //
-// The config is expected to be valid, ie slices deduped, all upper-case,
-// and only either IDs or Group/IncludeIDs/ExcludeIDs, with no overlap between
-// IncludeIDs and ExcludeIDs.
+// The configuration is expected to be valid, deduplicated, and all upper-case.
+// IncludeIDs and ExcludeIDs MUST NOT have an intersection.
```

Comment:
optional: We could add a validation step to ensure that `IncludeIDs` and `ExcludeIDs` are mutually exclusive and return an error. This would prevent users from relying on undefined behaviour.

Reply:
We actually already have this in place. You can see it in the current master here: https://github.com/uber/prototool/blob/dev/internal/settings/config_provider.go#L260